### PR TITLE
JUnit 5

### PR DESCRIPTION
**Pull request by litetex**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

* This PR drops JUnit 4 and introduces the new and improved JUnit 5.
* Minor code reworks and fixes on the fly

### CODE REVIEW

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/stream/YoutubeStreamExtractorDefaultTest.java [hunk 1]
```diff
@@ -59,54 +60,54 @@
...
             final StreamExtractor extractor =
                     YouTube.getStreamExtractor(BASE_URL + "8VajtrESJzA");
-            extractor.fetchPage();
+            assertThrows(PrivateContentException.class, extractor::fetchPage);
         }
 
-        @Test(expected = YoutubeMusicPremiumContentException.class)
-        public void youtubeMusicPremiumContent() throws Exception {
+        @Test
+        void youtubeMusicPremiumContent() throws Exception {
```
Comment (Stypox) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/stream/YoutubeStreamExtractorDefaultTest.java:107]: Why did you remove `public`?

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubePlaylistExtractorTest.java [hunk 2]
```diff
@@ -365,18 +366,18 @@ public void testUploaderUrl() throws Exception {
...
         public void testUploaderAvatarUrl() throws Exception {
             final String uploaderAvatarUrl = extractor.getUploaderAvatarUrl();
-            assertTrue(uploaderAvatarUrl, uploaderAvatarUrl.contains("yt"));
+            ExtractorAsserts.assertContains("yt", uploaderAvatarUrl);
         }
 
         @Test
         public void testStreamCount() throws Exception {
-            assertTrue("Error in the streams count", extractor.getStreamCount() > 40);
+            assertTrue(extractor.getStreamCount() > 40, "Error in the streams count");
```
Comment (Stypox) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubePlaylistExtractorTest.java:380]: Can't you use `ExtractorAsserts.assertAtLeast` here? Maybe also in other places.

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeMixPlaylistExtractorTest.java [hunk 3]
```diff
@@ -294,31 +292,37 @@ public void getStreamCount() {
...
             extractor = (YoutubeMixPlaylistExtractor) YouTube
                     .getPlaylistExtractor("https://www.youtube.com/watch?v=" + VIDEO_ID
                             + "&list=RD" + VIDEO_ID);
-            extractor.fetchPage();
-            extractor.getPage(new Page(""));
+
+            assertThrows(IllegalArgumentException.class, () -> {
+                extractor.fetchPage();
+                extractor.getPage(new Page(""));
+            });
```
Comment (Stypox) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeMixPlaylistExtractorTest.java:313]: Also here

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeMixPlaylistExtractorTest.java [hunk 4]
```diff
@@ -294,31 +292,37 @@ public void getStreamCount() {
...
             extractor = (YoutubeMixPlaylistExtractor) YouTube
                     .getPlaylistExtractor("https://www.youtube.com/watch?v=" + "abcde"
                             + "&list=RD" + "abcde");
-            extractor.fetchPage();
-            extractor.getName();
+
+            assertThrows(ExtractionException.class, () -> {
+                extractor.fetchPage();
+                extractor.getName();
+            });
```
Comment (Stypox) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeMixPlaylistExtractorTest.java:325]: I think `extractor.getName()` should be the only line included in the lambda, since it is the one that fails. If `fetchPage()` fails first, then `getName()` should be removed altogether

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelExtractorTest.java [hunk 5]
```diff
@@ -31,132 +31,125 @@
...
+        @Test
         public void noVideoTab() throws Exception {
             final ChannelExtractor extractor = YouTube.getChannelExtractor("https://invidio.us/channel/UC-9-kyTW8ZkZNDHQJ6FgpwQ");
-            extractor.fetchPage();
-            extractor.getInitialPage();
+
+            assertThrows(ContentNotSupportedException.class, () -> {
+                extractor.fetchPage();
+                extractor.getInitialPage();
+            });
```
Comment (Stypox) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelExtractorTest.java:145]: Also here

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelExtractorTest.java [hunk 6]
```diff
@@ -215,21 +208,21 @@ public void testMoreRelatedItems() throws Exception {
...
-            assertThat(extractor.getDescription(), containsString("Ungebremster Spieltrieb seit 1896."));
+            assertContains("Ungebremster Spieltrieb seit 1896.", extractor.getDescription());
         }
 
         @Test
         public void testAvatarUrl() throws Exception {
             String avatarUrl = extractor.getAvatarUrl();
             assertIsSecureUrl(avatarUrl);
-            assertTrue(avatarUrl, avatarUrl.contains("yt3"));
+            assertTrue(avatarUrl.contains("yt3"), avatarUrl);
```
Comment (Stypox) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelExtractorTest.java:218]: Why not use `assertContains` here, too? And in other places in this file

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelExtractorTest.java [hunk 7]
```diff
@@ -239,8 +232,8 @@ public void testFeedUrl() throws Exception {
 
         @Test
         public void testSubscriberCount() throws Exception {
-            assertTrue("Wrong subscriber count", extractor.getSubscriberCount() >= 0);
-            assertTrue("Subscriber count too small", extractor.getSubscriberCount() >= 4e6);
+            assertTrue(extractor.getSubscriberCount() >= 0, "Wrong subscriber count");
+            assertTrue(extractor.getSubscriberCount() >= 4e6, "Subscriber count too small");
```
Comment (Stypox) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelExtractorTest.java:236]: And here `assertAtLeast` and in other places in this file

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudSubscriptionExtractorTest.java [hunk 8]
```diff
@@ -61,7 +61,7 @@ public void testInvalidSourceException() {
                 // Ignore it, could be an unstable network on the CI server
             } catch (Exception e) {
                 boolean isExpectedException = e instanceof SubscriptionExtractor.InvalidSourceException;
-                assertTrue(e.getClass().getSimpleName() + " is not the expected exception", isExpectedException);
```
Comment (Stypox) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudSubscriptionExtractorTest.java:64]: Use `assertThrows` here, too?

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudPlaylistExtractorTest.java [hunk 9]
```diff
@@ -318,7 +320,8 @@ public void testUploaderAvatarUrl() {
 
         @Test
         public void testStreamCount() {
-            assertTrue("Stream count does not fit: " + extractor.getStreamCount(), extractor.getStreamCount() >= 370);
+            assertTrue(extractor.getStreamCount() >= 370,
```
Comment (Stypox) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudPlaylistExtractorTest.java:323]: `assertAtLeast` (& other places in this file)

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/peertube/PeertubeChannelExtractorTest.java [hunk 10]
```diff
@@ -233,7 +233,7 @@ public void testFeedUrl() throws ParsingException {
 
         @Test
         public void testSubscriberCount() throws ParsingException {
-            assertTrue("Wrong subscriber count", extractor.getSubscriberCount() >= 700);
+            assertTrue(extractor.getSubscriberCount() >= 700, "Wrong subscriber count");
```
Comment (Stypox) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/peertube/PeertubeChannelExtractorTest.java:236]: `assertAtLeast` (& other places in this file)

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/peertube/PeertubeAccountExtractorTest.java [hunk 11]
```diff
@@ -203,7 +202,7 @@ public void testFeedUrl() throws ParsingException {
 
         @Test
         public void testSubscriberCount() throws ParsingException {
-            assertTrue("Wrong subscriber count", extractor.getSubscriberCount() >= 100);
+            assertTrue(extractor.getSubscriberCount() >= 100, "Wrong subscriber count");
```
Comment (Stypox) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/peertube/PeertubeAccountExtractorTest.java:205]: `assertAtLeast` (& other places in this file)

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/media_ccc/MediaCCCConferenceListExtractorTest.java [hunk 12]
```diff
@@ -30,8 +30,8 @@ public static void setUpClass() throws Exception {
 
     @Test
     public void getConferencesListTest() throws Exception {
-        assertTrue("returned list was to small",
-                extractor.getInitialPage().getItems().size() >= 174);
+        assertTrue(extractor.getInitialPage().getItems().size() >= 174,
```
Comment (Stypox) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/media_ccc/MediaCCCConferenceListExtractorTest.java:33]: `assertAtLeast`

**Review by Stypox — CHANGES_REQUESTED** [review]
Great!

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/stream/YoutubeStreamExtractorDefaultTest.java (hunk 1 above)
Comment (litetex) [reply extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/stream/YoutubeStreamExtractorDefaultTest.java:107]: https://rules.sonarsource.com/java/RSPEC-5786

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudSubscriptionExtractorTest.java (hunk 8 above)
Comment (litetex) [reply extractor/src/test/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudSubscriptionExtractorTest.java:64]: Fixed and parameterized

File: extractor/src/main/java/org/schabi/newpipe/extractor/utils/Utils.java [hunk 13]
```diff
@@ -13,6 +13,10 @@
 
     public static final String HTTP = "http://";
     public static final String HTTPS = "https://";
+    /**
+     * @deprecated Use {@link java.nio.charset.StandardCharsets#UTF_8} instead
```
Comment (TobiGr) [inline extractor/src/main/java/org/schabi/newpipe/extractor/utils/Utils.java:17]: This is required for KitKat support. KitKat does not know `StandardCharset`.

Comment (opusforlife2) [reply extractor/src/main/java/org/schabi/newpipe/extractor/utils/Utils.java:17]: Since Kitkat support is being removed, shouldn't this be okay now?

Comment (litetex) [reply extractor/src/main/java/org/schabi/newpipe/extractor/utils/Utils.java:17]: > This is required for KitKat support. KitKat

Uhm what?
We are compiling against the Java and not Android.

Btw this was added in API 19 aka ``KitKat`` according to the docs so no problems here I guess 😄 
https://developer.android.com/reference/java/nio/charset/StandardCharsets

Comment (TobiGr) [reply extractor/src/main/java/org/schabi/newpipe/extractor/utils/Utils.java:17]: Ah sorry, for the legacy version, so < KitKat

Comment (litetex) [reply extractor/src/main/java/org/schabi/newpipe/extractor/utils/Utils.java:17]: > Ah sorry, for the legacy version, so < KitKat

Okay so let me clarify this: We can't use all Java 8 features - against which we are compiling - because we have to support an outdated > 9 year old Android version. And some person needs to tell you that otherwise the Extractor won't work anymore (on certain devices).
I wonder how this did even work so far and didn't result in major problems.
We should be definitely rework this. (Maybe by dropping the NP-Legacy - which is currently unmaintained).

I will revert the changes for now but this has to change.

Comment (opusforlife2) [reply extractor/src/main/java/org/schabi/newpipe/extractor/utils/Utils.java]: Is it not possible to have version specific code here? The `build.sdk.kitkat` thing.

Comment (Stypox) [reply extractor/src/main/java/org/schabi/newpipe/extractor/utils/Utils.java]: Doesn't desugaring magically solve this issue?

Comment (litetex) [reply extractor/src/main/java/org/schabi/newpipe/extractor/utils/Utils.java]: > Is it not possible to have version specific code here? The build.sdk.kitkat thing.

NewPipeExtractor is built on Java not Android. So no.

> Doesn't desugaring magically solve this issue?

Not sure about that one but I couldn't find anything inside https://developer.android.com/studio/write/java8-support-table.

**Comment by litetex** [comment]
@Stypox Ping

**Review by Stypox — APPROVED** [review]
Great! Tests pass and they have been well converted.
